Tests/UnitTests/SubscriberAttributes/BackendSubscriberAttributesTestBase.swift

```diff
@@ -231,86 +225,78 @@ class BackendSubscriberAttributesTestBase: XCTestCase {
             completionCallCount += 1
             receivedError = error
         })
 
         expect(self.mockHTTPClient.calls).toEventually(haveCount(1))
         expect(completionCallCount).toEventually(equal(1))
-        let receivedNSError = try XCTUnwrap(receivedError as NSError?)
 
-        expect(receivedNSError.code) == ErrorCode.networkError.rawValue
-        expect(receivedNSError.successfullySynced) == false
+        expect(receivedError?.successfullySynced) == false
+        expect(receivedError) == .networkError(underlyingError)
     }
 
     func testPostSubscriberAttributesSendsAttributesErrorsIfAny() throws {
         var completionCallCount = 0
 
+        let error: NetworkError = .errorResponse(
+            ErrorResponse.from([
+                Backend.RCAttributeErrorsKey: [
+                    [
+                        "key_name": "$some_attribute",
+                        "message": "wasn't valid"
+                    ]
+                ]
+            ]),
+            503
+        )
+
         self.mockHTTPClient.mock(
             requestPath: .postSubscriberAttributes(appUserID: appUserID),
-            response: .init(
-                error: ErrorResponse
-                    .from([
-                        Backend.RCAttributeErrorsKey: [
-                            [
-                                "key_name": "$some_attribute",
-                                "message": "wasn't valid"
-                            ]
-                        ]
-                    ])
-                    .asBackendError(with: 503)
-            )
+            response: .init(error: error)
         )
 
-        var receivedError: Error?
+        var receivedError: BackendError?
         backend.post(subscriberAttributes: [
             subscriberAttribute1.key: subscriberAttribute1,
             subscriberAttribute2.key: subscriberAttribute2
         ],
                      appUserID: appUserID,
-                     completion: { (error: Error!) in
+                     completion: {
             completionCallCount += 1
-            receivedError = error
+            receivedError = $0
         })
 
         expect(self.mockHTTPClient.calls).toEventually(haveCount(1))
         expect(completionCallCount).toEventually(equal(1))
+        expect(receivedError).toEventuallyNot(beNil())
 
-        let receivedNSError = try XCTUnwrap(receivedError as NSError?)
-        expect(receivedNSError.code) == ErrorCode.unknownBackendError.rawValue
-        expect(receivedNSError.userInfo[Backend.RCAttributeErrorsKey]).toNot(beNil())
-
-        let receivedAttributeErrors = receivedNSError.userInfo[Backend.RCAttributeErrorsKey]
-        guard let receivedAttributeErrors = receivedAttributeErrors as? [String: String] else {
-            fail("received attribute errors are not of type [String: String]")
-            return
-        }
-        expect(receivedAttributeErrors) == ["$some_attribute": "wasn't valid"]
```

Comment:
All this is unnecessary here, it's already tested as part of the `NetworkError` -> `NSError` conversion (see `NetworkErrorTests.testErrorResponse`.